src/main/js/oauth.js

```diff
@@ -292,6 +292,15 @@ gromit.oAuthAuthenticate = function(/*String*/ url, /*function*/ callback, /*boo
     url += '?response_type=token';
     url += '&redirect_uri=' + gromit.getCurrentUrl() + 'oauth.html';
     url += '&client_id=' + gromit.ClientId;
+    
+    if (typeof gromit.AuthParams !== 'undefined') {
+        for (var i = 0; i < gromit.AuthParams.length; i++) {
+            url += '&' + gromit.AuthParams[i].name;
+            if (gromit.AuthParams[i].value !== 'null') {
+                url += '=' + gromit.AuthParams[i].value;
```

Comment:
Do we need to URL encode these values?

Reply:
No more nor less than the redirect_uri, client_id or anything else that could have reserved characters. I figured that the supplier of data (i.e., the server side) could handle that if necessary.